samtranslator/plugins/api/implicit_api_plugin.py

```diff
@@ -195,6 +204,32 @@ def _add_api_to_swagger(self, event_id, event_properties, template):
         resource.properties["DefinitionBody"] = editor.swagger
         template.set(api_id, resource)
 
+    def _maybe_add_condition_to_implicit_api(self, template_dict):
+        """
+        Decides whether to add a condition to the implicit api resource.
+        :param dict template_dict: SAM template dictionary
+        """
+        if None not in self.conditions and len(self.conditions) > 0:
+            implicit_api_resource = template_dict.get('Resources').get(self.implicit_api_logical_id)
+            if len(self.conditions) == 1:
+                condition = self.conditions.pop()
+                implicit_api_resource['Condition'] = condition
+            else:
+                implicit_api_resource['Condition'] = self.implicit_api_condition
+                self._add_condition_to_template(template_dict)
+
+    def _add_condition_to_template(self, template_dict):
+        """
+        Adds necessary conditions to the template for the implicit api
+        :param dict template_dict: SAM template dictionary
+        """
+        if not template_dict.get('Conditions'):
+            template_dict['Conditions'] = {}
```

Comment:
By default, if Conditions exist on these resources, they should also exist in the Conditions section of the template. Should I remove this check?